src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -8314,6 +8336,13 @@ namespace VariableSpeedCoils {
             varSpeedCoil.RunFrac = 0.0;
             varSpeedCoil.PartLoadRatio = 0.0;
 
+            int inNode = varSpeedCoil.AirInletNodeNum;
+            if (inNode > 0) {
+                varSpeedCoil.InletAirDBTemp = state.dataLoopNodes->Node(inNode).Temp;
```

Comment:
Why are these needed here? And seems like `inNode` must be > 0 if it's gotten this far.

Also, down at line 8387, I see `CO2` is passed through, but I don't see the same for `GenContam` (hmm, not for this PR, but noting this for later).

Reply:
@mjwitte On second thought, I agree with you that this isn't needed.  I think I was concerned that if the user zeroed out the capacity and then messed with the nodes that inNode could end up being zero.  But the reality is that isn't going to happen and a bad inNode will get flagged earlier on.  I will back these changes out as they aren't necessary.

Reply:
@mjwitte Do you want to log the GenContam issue as a new defect...or was that a request for me to do it?

Reply:
@mjwitte Oh, now I remember why I put this in.  Because if it skips the Init routine as I previously proposed, it never actually sets the information at the component Inlet leaving everything zero--which was not correct.  So, I had added this to resolve that issue.  I can back this out though because with Rich's proposed solution, it will still go through the Init routine and set things at the Inlet properly.

Reply:
> @mjwitte Do you want to log the GenContam issue as a new defect...or was that a request for me to do it?

I'll take care of it. I'll be opening another PR related to CO2 and GenContam so I'll probably just fix it there.
https://github.com/NREL/EnergyPlus/issues/10473